core/store/src/trie/state_snapshot.rs

```diff
@@ -134,11 +134,47 @@ impl StateSnapshot {
 }
 
 /// Information needed to make a state snapshot.
-#[derive(Default)]
-pub struct StateSnapshotConfig {
-    pub home_dir: PathBuf,
-    pub hot_store_path: PathBuf,
-    pub state_snapshot_subdir: PathBuf,
+#[derive(Default, Debug)]
+pub enum StateSnapshotConfig {
+    #[default]
+    Disabled,
+    Enabled {
+        state_snapshots_dir: PathBuf,
+    },
```

Comment:
enabled must be the default

Reply:
Sure, will change.

Reply:
Does it make sense though to have a default option that is non-usable? I mean that if enabled is default, it will contain an empty `PathBuf`... 🤔

Reply:
As discussed offline, I'll remove the `Default` altogether.